CONTRIBUTING.md

VanshBhardwaj24 opened the pull request:
Created contributing guidelines

## What does this PR do?

Created contributing markdown file, fixed the issue and have created a pull request for the same.

Fixes #1760

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update


## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code and corrected any misspellings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

deepakprabhakara commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,129 @@
+
+## License
+
+Jackson is an open-source project released under the [MIT License](LICENSE). Your contributions are subject to the terms of this license.
```
Apache 2.0 license, not MIT

deepakprabhakara commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,129 @@
+
+### Code of Conduct
+
+Please ensure that your contributions align with our https://github.com/boxyhq/saas-starter-kit/blob/main/CODE_OF_CONDUCT.md. Be respectful, inclusive, and considerate of others.
```
Wrong url here

VanshBhardwaj24 commented:
Hey, i have changed the license and remove code of conduct since there
wasn't one for it yet .

Thanks!

On Thu, Oct 12, 2023 at 4:44 PM Deepak Prabhakara ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In CONTRIBUTING.md
> <https://github.com/boxyhq/jackson/pull/1766#discussion_r1356665920>:
>
> > +## Development
> +
> +1. Fork the repository.
> +2. Clone your fork to your local environment.
> +3. Install dependencies.
> +4. Implement changes, add features, or enhance documentation.
> +5. Test your changes.
> +6. Submit a pull request against the main branch.
> +
> +## License
> +
> +Jackson is an open-source project released under the [MIT License](LICENSE). Your contributions are subject to the terms of this license.
> +
> +### Code of Conduct
> +
> +Please ensure that your contributions align with our https://github.com/boxyhq/saas-starter-kit/blob/main/CODE_OF_CONDUCT.md. Be respectful, inclusive, and considerate of others.
>
> Wrong url here
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/boxyhq/jackson/pull/1766#pullrequestreview-1673900726>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASJZT43ULJUEYTTDW63OFXTX67GK3ANCNFSM6AAAAAA55IEZGA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

VanshBhardwaj24 commented:
Added code of conduct as well linked the github links for the above ,thanks.


On Thu, Oct 12, 2023 at 9:18 PM Vansh ***@***.***> wrote:

> Hey, i have changed the license and remove code of conduct since there
> wasn't one for it yet .
>
> Thanks!
>
> On Thu, Oct 12, 2023 at 4:44 PM Deepak Prabhakara <
> ***@***.***> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In CONTRIBUTING.md
>> <https://github.com/boxyhq/jackson/pull/1766#discussion_r1356665920>:
>>
>> > +## Development
>> +
>> +1. Fork the repository.
>> +2. Clone your fork to your local environment.
>> +3. Install dependencies.
>> +4. Implement changes, add features, or enhance documentation.
>> +5. Test your changes.
>> +6. Submit a pull request against the main branch.
>> +
>> +## License
>> +
>> +Jackson is an open-source project released under the [MIT License](LICENSE). Your contributions are subject to the terms of this license.
>> +
>> +### Code of Conduct
>> +
>> +Please ensure that your contributions align with our https://github.com/boxyhq/saas-starter-kit/blob/main/CODE_OF_CONDUCT.md. Be respectful, inclusive, and considerate of others.
>>
>> Wrong url here
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/boxyhq/jackson/pull/1766#pullrequestreview-1673900726>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ASJZT43ULJUEYTTDW63OFXTX67GK3ANCNFSM6AAAAAA55IEZGA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

VanshBhardwaj24 commented:
@deepakprabhakara sir please review and close it or let me know for any changes since there wasn't clear description of this issue and i had to search througth other refrences and see documentation on this.

deepakprabhakara commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,143 @@
+
+## Code Of Conduct 
+
+Please ensure that your contributions align with our [Code of Conduct](https://github.com/boxyhq/jackson/blob/6cf5639706714c727d093136372a46ac7b26ac0a/CODE_OF_CONDUCT.md#L1). Be respectful, inclusive, and considerate of others.
```
Please point it to main branch and without the L1 anchor

deepakprabhakara replied:
@VanshBhardwaj24 This one is the last pending issue

VanshBhardwaj24 replied:
@deepakprabhakara fixed this and thank you so much for your guidance as i am beginner

deepakprabhakara commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,143 @@
+
+```shell
+npm run dev
+cp .env.example .env
````
cp should come before the npm run dev command

deepakprabhakara reviewed (changes requested):
One last minor thing to fix (the URL for Code of Conduct)

deepakprabhakara commented on CONTRIBUTING.md:
````diff
@@ -31,14 +31,14 @@ Navigate to the project folder and install necessary dependencies:
 ```shell
 cd jackson
-npm run custom-install
+npm install
````
`npm run custom-install` is correct, we have some legacy peer dependencies which fail with `npm install`

VanshBhardwaj24 replied:
This is my first Pull request i have ever made please ignore all this mistakes sir

VanshBhardwaj24 commented:
> One last minor thing to fix (the URL for Code of Conduct)

fixed that @deepakprabhakara and pointed it to main branch ,thanks

deepakprabhakara commented:
@VanshBhardwaj24 Can you please keep it consistent with https://github.com/boxyhq/saas-starter-kit/blob/main/CONTRIBUTING.md, latest commit makes changes that diverges from this.

VanshBhardwaj24 commented:
> @VanshBhardwaj24 Can you please keep it consistent with https://github.com/boxyhq/saas-starter-kit/blob/main/CONTRIBUTING.md, latest commit makes changes that diverges from this.

Done !

deepakprabhakara commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,161 @@
+### Make and commit your changes with a note
+
+```shell
+git commit -m ""enter the message for changes that need to be committed"
````
git commit -m "enter the message for changes that need to be committed"

VanshBhardwaj24 replied:
Sorry for such typos i have corrected all the typos and logical errors,thanks @deepakprabhakara

deepakprabhakara commented:
Thanks @VanshBhardwaj24
